docs/troubleshooting_guide.md

```diff
@@ -0,0 +1,358 @@
+
+#### 8. Time offset between RAC cluster nodes
+* Ansible error:
+```bash
+"[FATAL] [INS-13019] Some mandatory prerequisites are not met. These prerequisites cannot be ignored.", 
+"   CAUSE: The following prerequisites are mandatory and cannot be ignored: ", "- Time offset between nodes", "  
+```
+
+* Fix:
+  * RAC nodes need an NTP server to synchronize time.
+  * In the case that there are no existing NTP servers, it is recommended to use the cloud VM as the NTP server.
+  * The following flag may be with `install-oracle.sh` to provide the name of the NTP server that needs to be used:
+```bash
+--ntp-pref 
+```
+
+  * The above will mimic following manual fix:
+
+```bash
+[root@control-host-cloudvm ~]# #Add this longer line in cloud VM host:
+[root@control-host-cloudvm ~]# diff  /etc/ntp.conf /etc/ntp.conf.bk.jc.01.21.2021 
+62c62
+< restrict 192.16.0.0 mask 255.255.0.0 nomodify notrap
+---
+> restrict 192.16.0.0 mask 255.255.0.0
+```
+
+```bash
+[root@control-host-cloudvm ~]# #restart ntpd:
+[root@control-host-cloudvm ~]# systemctl restart ntpd
+```
+
+```bash
+[root@racnode1 ~]# #Add this in both BM servers to instruct the OS to reach out to the VM jump host as a NTP server:
+[root@racnode1 ~]# diff /etc/ntp.conf /etc/ntp.conf.bk.jc.01.21.2021
+21d20
+< server 10.210.1.7 prefer
+```
+
+
+```bash
+# After the fix
+[janedoe@control-host-cloudvm ~]$ date ; ssh 172.16.30.1 date ; ssh 172.16.30.2 date
+Thu Jan 21 22:40:29 UTC 2021
+Thu Jan 21 22:40:31 UTC 2021
+Thu Jan 21 22:40:34 UTC 2021
+```
+
+#### 9. Pass extra flags into ansible via `install-oracle.sh`:
+* There may arise situations where we may want to pass extra flags to ansible and this may be done as follows (taking an example of passing the debug flag):
+```bash
+ ~/git-clone-holder/bms-toolkit [master L|✚ 3…2] 
+23:17 $ ./install-oracle.sh \ 
+--ora-swlib-bucket gs://bmaas-testing-oracle-software \ 
+--instance-ssh-user ansible \ 
+--instance-ssh-key ~/.ssh/id_rsa \ 
+--backup-dest /u01/backups \ 
+--ora-swlib-path /u01/oracle_install \ 
+--ora-version 19 \ 
+--ora-swlib-type gcs \ 
+--ora-asm-disks asm_disk_config.json \ 
+--ora-data-mounts data_mounts_config.json \ 
+--cluster-type RAC \ 
+--cluster-config cluster_config.json \ 
+--ora-data-diskgroup DATA \ 
+--ora-reco-diskgroup RECO \ 
+--ora-db-name orcl \ 
+--ora-db-container false >> ~/git-clone-holder/bms-toolkit/logs/sydney-1.out \ 
+-- -vvvv 2>&1   <-----------------------------------------------------------------------------
+```
+
+* `-- -vvvv` will be sent by the wrapper install-oracle.sh into Ansible as: 
+```bash
+Ansible params: -vvvv
+Found Ansible at /usr/bin/ansible-playbook
+
+Running Ansible playbook: /usr/bin/ansible-playbook -i ./inventory_files/inventory_orcl_RAC  -vvvv check-instance.yml
+```
+
+* Debug tip, if you are testing a **specific functionality** of a playbook by extracting snippets of the playbook into your local ansible setup, don't turn off the `gather_facts`.
```

Comment:
Not sure what this means;  how does one turn gather_facts on or off?

Reply:
Added context